index.js

```diff
@@ -224,6 +225,11 @@ module.exports = class Hyperswarm extends EventEmitter {
       if (!opened) this._connectDone()
       this.stats.connects.client.closed++
 
+      const err = getStreamError(conn)
+      if (shouldBan(err)) {
```

Comment:
I'd emit an event and track in a stat how many times we ban peers, that way we have optional visibility on it. I can make a follow-up PR for that